Examples/Docs/Sources/AsyncExamples/main.swift

```diff
@@ -246,34 +247,94 @@ private func versionedAPI() throws {
-            // error
-            print(error)
-        }
-
         // End Versioned API Example 2
-        try? cursorFuture.wait()
+
         try client.syncClose()
     }
 
     do {
         // Start Versioned API Example 3
 
-        // Use the `deprecationErrors` option
-        var opts = MongoClientOptions()
-        opts.serverAPI = MongoServerAPI(version: .v1, deprecationErrors: true)
+        let opts = MongoClientOptions(
+            serverAPI: MongoServerAPI(version: .v1, strict: false)
+        )
+        let client = try MongoClient(uri, using: myEventLoopGroup, options: opts)
+
+        // End Versioned API Example 3
+
+        try client.syncClose()
+    }
+
+    do {
+        // Start Versioned API Example 4
+
+        let opts = MongoClientOptions(
+            serverAPI: MongoServerAPI(version: .v1, deprecationErrors: true)
+        )
         let client = try MongoClient(uri, using: myEventLoopGroup, options: opts)
+
         // End Versioned API Example 3
 
         try client.syncClose()
     }
 }
+
+private func versionedAPIMigrationExample() throws {
+    let myEventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+    let options = MongoClientOptions(
+        serverAPI: MongoServerAPI(version: .v1, strict: true)
+    )
+    let client = try MongoClient(using: myEventLoopGroup, options: options)
+    try client.db("test").drop().wait()
+
+    let dateFormatter = ISO8601DateFormatter()
+    dateFormatter.formatOptions = [.withInternetDateTime]
+
+    // swiftlint:disable line_length
+    // 1. Populate a test sales collection
+    let insertResult = client.db("test").collection("sales").insertMany([
+        ["_id": 1, "item": "abc", "price": 10, "quantity": 2, "date": .datetime(dateFormatter.date(from: "2021-01-01T08:00:00Z")!)],
+        ["_id": 2, "item": "jkl", "price": 20, "quantity": 1, "date": .datetime(dateFormatter.date(from: "2021-02-03T09:00:00Z")!)],
+        ["_id": 3, "item": "xyz", "price": 5, "quantity": 5, "date": .datetime(dateFormatter.date(from: "2021-02-03T09:00:00Z")!)],
+        ["_id": 4, "item": "abc", "price": 10, "quantity": 10, "date": .datetime(dateFormatter.date(from: "2021-02-15T08:00:00Z")!)],
+        ["_id": 5, "item": "xyz", "price": 5, "quantity": 10, "date": .datetime(dateFormatter.date(from: "2021-02-15T09:05:00Z")!)],
+        ["_id": 6, "item": "xyz", "price": 5, "quantity": 5, "date": .datetime(dateFormatter.date(from: "2021-02-15T12:05:10Z")!)],
+        ["_id": 7, "item": "xyz", "price": 5, "quantity": 10, "date": .datetime(dateFormatter.date(from: "2021-02-15T14:12:12Z")!)],
+        ["_id": 8, "item": "abc", "price": 10, "quantity": 5, "date": .datetime(dateFormatter.date(from: "2021-03-16T20:20:13Z")!)]
+    ])
+    // End step 1
+    // swiftlint:enable line_length
+
+    try insertResult.wait()
+
+    // 2. Run "count" using a strict client, observe error
+    let countResult = client.db("test").runCommand(["count": "sales"])
```

Comment:
do we want to construct this as a callback on `insertResult`? I worry that a user might copy/paste the examples and have this and the insert running concurrently. ditto for the `countDocuments` one.

Reply:
good point; I updated both calls to wait on the insert